projects/Mallard/src/authentication/__tests__/deep-link-auth.spec.ts

```diff
@@ -32,87 +34,89 @@ describe('deep-link-auth', () => {
         describe('external link', () => {
             it('waits for a deep link with a token before resolving', async () => {
                 const linking = Object.assign(createListener(), {
-                    openURL: () => {},
+                    openURL: () => {
+                        // Only emit the url once we actually opened the
+                        // original. Must be asynchronous for consistency with
+                        // real behavior.
+                        setImmediate(() => {
+                            linking.emit('url', { url: 'myurl' })
+                        })
+                    },
                 })
                 const appState = createListener()
                 const validator = jest.fn(async () => 'token')
 
-                const promise = authWithDeepRedirect(
+                const val = await authWithDeepRedirect(
                     'https://authurl.com/auth',
+                    'myurl',
```

Comment:
```suggestion
                    'socialurl',
```

Reply:
Sure. I was just keeping the existing one (see below), but you're right there are better choices